### PR TITLE
TLS

### PR DESCRIPTION
**Pull request by luckyraul**
@garethr I am not confident for supporting all OS, but it should be fine

### CODE REVIEW

**Comment by JWGmeligMeyling** [comment]
Hi @luckyraul, the solution looks promising to me. I think this PR can be improved by adding a test and information to the README on how to use / generate these certificates.

File: manifests/init.pp [hunk 1]
```diff
@@ -423,6 +449,13 @@
     fail('Values for dm_ variables will be ignored unless storage_driver is set to devicemapper.')
   }
 
+  if($tls_enable) {
```
Comment (garethr) [inline manifests/init.pp]: It would also be nice to warn if `tls_*` are set but `tls_enable` is set to false, as they then get silently ignored.

Comment (garethr) [inline manifests/init.pp]: Actually, these have sane defaults so this is probably fine.

**Comment by garethr** [comment]
This looks great. Thanks @luckyraul. It would be really useful to have some simple unit tests added to the existing test suite to verify this, and help make sure it doesn't get accidentally broken. The CONTRIBUTING doc has some details but please do ask any questions if you're not too familiar with Puppet testing.

**Comment by JWGmeligMeyling** [comment]
@luckyraul Thanks for adding the tests, and the build passes now as well. @garethr Are the unit tests sufficient or should Beaker integration tests also be added to ensure Docker indeed uses TLS now? Or will we test this manually?

**Comment by garethr** [comment]
@luckyraul this looks great. Thanks for adding the tests as well.

**Pull request merged**